index.js

```diff
@@ -1,145 +1,143 @@
-var events = require('events')
-var inherits = require('inherits')
-
-module.exports = LRU
-
-function LRU (opts) {
-  if (!(this instanceof LRU)) return new LRU(opts)
-  if (typeof opts === 'number') opts = {max: opts}
-  if (!opts) opts = {}
-  events.EventEmitter.call(this)
-  this.cache = {}
-  this.head = this.tail = null
-  this.length = 0
-  this.max = opts.max || 1000
-  this.maxAge = opts.maxAge || 0
-}
-
-inherits(LRU, events.EventEmitter)
+const EventEmitter = require('events')
+
+class LRU extends EventEmitter {
+  constructor (opts) {
+    super()
+    if (typeof opts === 'number') opts = { max: opts }
+    if (!opts) opts = {}
+    this.cache = {}
```

Comment:
Could be a Map? Also I don't think this will get merged as lib seems stalled/dead. Worth forking and updating if that's the case

Reply:
Object are faster?
could do `this.cache = Object.create(null)` also

Depends on if you want to have other type of keys... like `x.set(window, 'abc')`

if i would have forked it then i would probably also remove the `EventEmitter` for better cross environment support without requiring node only stuff and use something like: `this.onEvict = fn` or perhaps remove it entirely and use WeakRef if you really need to know when it's GC'ed

Reply:
or used the global EventTarget on node v15